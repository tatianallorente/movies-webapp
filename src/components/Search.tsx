import React, { useState, useContext } from "react";

import { Box, Button, TextField } from "@mui/material";

import { SearchContext } from "../context";

const Search = () => {
	const [searchForm, setSearchForm] = useState("");

	const { updateSearchQuery, resetSearchQuery } = useContext(SearchContext);

	const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
		setSearchForm(e.target.value);
	};

	const handleSearch = (e: React.FormEvent<HTMLFormElement>) => {
		e.preventDefault();
		updateSearchQuery(searchForm);
	};

	const resetFormAndCleanSearch = () => {
		setSearchForm("");
		resetSearchQuery();
	};

	return (
		<Box
			component="form"
			autoComplete="off"
			onSubmit={handleSearch}
			mt={4}
			mb={4}
			display="flex"
			justifyContent="center"
			alignItems="stretch"
			gap={1}
		>
			<TextField
				id="searchForm"
				name="searchForm"
				placeholder="Avengers, Matrix..."
				variant="outlined"
				size="small"
				value={searchForm}
				onChange={handleInputChange}
			/>
			<Button
				variant="contained"
				type="submit"
				disableElevation
				color="secondary"
			>
				Search
			</Button>
			<Button
				variant="outlined"
				type="button"
				color="secondary"
				onClick={resetFormAndCleanSearch}
			>
				Reset
			</Button>
		</Box>
	);
};

export default Search;
