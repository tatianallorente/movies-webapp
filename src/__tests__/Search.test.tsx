import { fireEvent, render } from "@testing-library/react";
import Search from "../components/Search";

describe("Tests in component <Search />", () => {
	const searchQuery = "Avengers";

	test("Search component is rendered", () => {
		const { container } = render(<Search />);

		expect(container).toMatchSnapshot();
	});

	test("Search input field updates state correctly", () => {
		const { getByPlaceholderText } = render(<Search />);
		const input = getByPlaceholderText("Avengers, Matrix...") as HTMLInputElement;
		fireEvent.change(input, { target: { value: "Avengers" } });
		expect(input.value).toBe("Avengers");
	});

	test("Reset button calls resetSearchQuery function and clears input field", () => {
		const { getByPlaceholderText, getByRole } = render(<Search />);
		const input = getByPlaceholderText("Avengers, Matrix...") as HTMLInputElement;
		const resetButton = getByRole("button", { name: "Reset" });

		fireEvent.change(input, { target: { value: searchQuery } });
		expect(input.value).toBe("Avengers");

		fireEvent.click(resetButton);
		expect(input.value).toBe("");
	});
});
