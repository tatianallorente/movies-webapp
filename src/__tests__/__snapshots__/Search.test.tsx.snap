// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in component <Search /> Search component is rendered 1`] = `
<div>
  <form
    autocomplete="off"
    class="MuiBox-root css-y0wmzg"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSuccess MuiInputBase-formControl MuiInputBase-sizeSmall css-kzf2d7-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
          id="searchForm"
          name="searchForm"
          placeholder="Avengers, Matrix..."
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-1p4a22o-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="submit"
    >
      Search
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1cbixbc-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Reset
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </form>
</div>
`;
